hypervisor/arch/x86/configs/vm_config.c

```diff
@@ -119,35 +119,24 @@ static bool check_vm_clos_config(uint16_t vm_id)
 bool sanitize_vm_config(void)
 {
 	bool ret = true;
-	uint16_t vm_id, vcpu_id, vuart_idx, nr;
-	uint64_t sos_pcpu_bitmap, pre_launch_pcpu_bitmap = 0U, vm_pcpu_bitmap;
+	uint16_t vm_id, vuart_idx;
+	uint64_t pre_launch_pcpu_bitmap = 0UL;
 	struct acrn_vm_config *vm_config;
 
-	sos_pcpu_bitmap = (uint64_t)((((uint64_t)1U) << get_pcpu_nums()) - 1U);
 	/* All physical CPUs except ocuppied by Pre-launched VMs are all
-	 * belong to SOS_VM. i.e. The vm_pcpu_bitmap of a SOS_VM is decided
-	 * by vm_pcpu_bitmap status in PRE_LAUNCHED_VMs.
+	 * belong to SOS_VM. i.e. The pcpu_affinity_bitmap of a SOS_VM is decided
+	 * by pcpu_affinity_bitmap status in PRE_LAUNCHED_VMs.
 	 * We need to setup a rule, that the vm_configs[] array should follow
 	 * the order of PRE_LAUNCHED_VM first, and then SOS_VM.
 	 */
 	for (vm_id = 0U; vm_id < CONFIG_MAX_VM_NUM; vm_id++) {
 		vm_config = get_vm_config(vm_id);
-		vm_pcpu_bitmap = 0U;
-		if (vm_config->load_order != SOS_VM) {
-			for (vcpu_id = 0U; vcpu_id < vm_config->vcpu_num; vcpu_id++) {
-				if (bitmap_weight(vm_config->vcpu_affinity[vcpu_id]) != 1U) {
-					pr_err("%s: vm%u vcpu%u should have only one prefer pcpu!",
-							__func__, vm_id, vcpu_id);
-					ret = false;
-					break;
-				}
-				vm_pcpu_bitmap |= vm_config->vcpu_affinity[vcpu_id];
-			}
+		vm_config->vcpu_num = bitmap_weight(vm_config->pcpu_affinity_bitmap);
 
-			if ((bitmap_weight(vm_pcpu_bitmap) != vm_config->vcpu_num)) {
-				pr_err("%s: One VM cannot have multiple vcpus share one pcpu!", __func__);
-				ret = false;
-			}
+		if ((vm_config->pcpu_affinity_bitmap & ~((1UL << get_pcpu_nums()) - 1UL)) != 0UL) {
```

Comment:
We have definition of ALL_CPUS_MASK. Use this MACRO?

Reply:
BTW, we have ALL_CPUS_MASK and AP_MASK, they are in different H files. Can you move them to same file?
Also AP_MASK should be defined as  (ALL_CPUS_MASK &  ~(1U << BSP_CPU_ID))

Reply:
Sure, I can move both of them to cpu.h

and will change the following to 64 bits.
 #define ALL_CPUS_MASK           (uint32_t) (((uint32_t)1U << (uint32_t) get_pcpu_nums()) - (uint32_t)1U)